source/slang/slang-check-decl.cpp

```diff
@@ -11582,6 +11567,139 @@ int compareDecls(ASTBuilder* astBuilder, Decl& lhs, Decl& rhs, int* feedback)
     return res;
 }
 
+int compareDeclRefBases(ASTBuilder* astBuilder, DeclRefBase& lhs, DeclRefBase& rhs, int* feedback)
```

Comment:
make this `compareDeclRef`, and compare the decl first in the function.
If decl is the same, continue with the rest of the logic here.